dev/io.openliberty.jakarta.validation.v31_fat/test-applications/val31/src/val31/web/Validation31TestServlet.java

```diff
@@ -62,19 +64,30 @@ public void recordValidatePropertyAndValueTest() throws Exception {
     }
 
     /**
-     * Test that the result of a method on a record can be validated.
+     * This tests if able to validate parameters and return value with records.
      */
-
     @Test
-    public void recordMethodValidationTest() throws Exception {
-        // TODO Add a method that includes validation to a record class and confirm that it can be validated
+    public void validateRecordParameters() throws Exception {
 
         Person object = new Person("x");
-        Method method = Person.class.getMethod("getName");
-        String returnValue = object.getName();
+        Method method = Person.class.getMethod("checkNameSize", String.class);
+
+        Object[] parameterValues = { "valuetocheck" };
+        Set<ConstraintViolation<Person>> violations1 = validator.forExecutables()
+                        .validateParameters(object, method,
+                                            parameterValues);
+        assertEquals(1, violations1.size());
+        Class<? extends Annotation> constraintType = violations1.iterator()
+                        .next()
+                        .getConstraintDescriptor()
+                        .getAnnotation()
+                        .annotationType();
+        assertEquals(Size.class, constraintType);
+        String returnvalue = object.checkNameSize("x");
```

Comment:
This should still be using `getName` to test the return value of the method

Reply:
Updated all the comments